libs/coin-modules/coin-cardano/src/buildTransaction.ts

```diff
@@ -44,28 +42,34 @@ function getTyphonInputFromUtxo(utxo: CardanoOutput): TyphonTypes.Input {
   };
 }
 
-function getRewardWithdrawalCertificate(account: CardanoAccount): TyphonTypes.Withdrawal | null {
-  if (!account.cardanoResources.delegation?.rewards.gt(0)) {
-    return null;
-  }
-
-  const stakeCredential = getAccountStakeCredential(account.xpub as string, account.index);
-  const stakeKeyHashCredential: TyphonTypes.HashCredential = {
-    hash: Buffer.from(stakeCredential.key, "hex"),
-    type: TyphonTypes.HashType.ADDRESS,
-    bipPath: stakeCredential.path,
-  };
-
-  const networkId = isTestnet(account.currency)
-    ? TyphonTypes.NetworkId.TESTNET
-    : TyphonTypes.NetworkId.MAINNET;
-  const rewardAddress = new RewardAddress(networkId, stakeKeyHashCredential);
-  const rewardsWithdrawalCertificate: TyphonTypes.Withdrawal = {
-    rewardAccount: rewardAddress,
-    amount: account.cardanoResources.delegation.rewards,
-  };
-
-  return rewardsWithdrawalCertificate;
+function getRewardWithdrawalCertificate(_account: CardanoAccount): TyphonTypes.Withdrawal | null {
+  return null;
+
+  /**
+   * Disable rewards withdraw certificate, as a work around for Chang 2 hard fork
+   */
+
+  // if (!account.cardanoResources.delegation?.rewards.gt(0)) {
+  //   return null;
+  // }
+
+  // const stakeCredential = getAccountStakeCredential(account.xpub as string, account.index);
+  // const stakeKeyHashCredential: TyphonTypes.HashCredential = {
+  //   hash: Buffer.from(stakeCredential.key, "hex"),
```

Comment:
Don't you want to delete all those comments ?

Reply:
Hi @qperrot, this changes are temporary and that commented code will be in use later 🙂